pkg/blockchain/blockchainPublisher.go

```diff
@@ -62,46 +64,92 @@ func (m *BlockchainPublisher) PublishGroupMessage(
 	ctx context.Context,
 	groupID [32]byte,
 	message []byte,
-) (common.Hash, error) {
+) (*abis.GroupMessagesMessageSent, error) {
+	if len(message) == 0 {
+		return nil, errors.New("message is empty")
+	}
 	tx, err := m.messagesContract.AddMessage(&bind.TransactOpts{
 		Context: ctx,
 		From:    m.signer.FromAddress(),
 		Signer:  m.signer.SignerFunc(),
 	}, groupID, message)
 	if err != nil {
-		return common.Hash{}, err
+		return nil, err
 	}
 
-	return WaitForTransaction(
+	receipt, err := WaitForTransaction(
 		ctx,
 		m.logger,
 		m.client,
 		2*time.Second,
 		250*time.Millisecond,
 		tx.Hash(),
 	)
+	if err != nil {
+		return nil, err
+	}
+
+	if receipt == nil {
+		return nil, errors.New("transaction receipt is nil")
+	}
+
+	return findLog(receipt, m.messagesContract.ParseMessageSent, "no message sent log found")
 }
 
 func (m *BlockchainPublisher) PublishIdentityUpdate(
 	ctx context.Context,
 	inboxId [32]byte,
 	identityUpdate []byte,
-) (common.Hash, error) {
+) (*abis.IdentityUpdatesIdentityUpdateCreated, error) {
+	if len(identityUpdate) == 0 {
+		return nil, errors.New("identity update is empty")
+	}
 	tx, err := m.identityUpdateContract.AddIdentityUpdate(&bind.TransactOpts{
 		Context: ctx,
 		From:    m.signer.FromAddress(),
 		Signer:  m.signer.SignerFunc(),
 	}, inboxId, identityUpdate)
 	if err != nil {
-		return common.Hash{}, err
+		return nil, err
 	}
 
-	return WaitForTransaction(
+	receipt, err := WaitForTransaction(
 		ctx,
 		m.logger,
 		m.client,
 		2*time.Second,
 		250*time.Millisecond,
 		tx.Hash(),
 	)
+	if err != nil {
+		return nil, err
+	}
+	if receipt == nil {
+		return nil, errors.New("transaction receipt is nil")
+	}
+
+	return findLog(
+		receipt,
+		m.identityUpdateContract.ParseIdentityUpdateCreated,
+		"no identity update log found",
+	)
+}
+
+func findLog[T any](
+	receipt *types.Receipt,
+	parse func(types.Log) (*T, error),
+	errorMsg string,
+) (*T, error) {
+	for _, logEntry := range receipt.Logs {
+		if logEntry == nil {
+			continue
+		}
+		event, err := parse(*logEntry)
+		if err != nil {
+			continue
+		}
```

Comment:
Can there be multiple logs for the same update type, e.g. submitted by others?

Reply:
The receipt is scoped to a single transaction, so it will only be your own events.